platform-core-module/src/main/java/com/softicar/platform/core/module/email/mailbox/imap/ImapConnection.java

```diff
@@ -1,30 +1,30 @@
-package com.softicar.platform.core.module.email.mailbox;
+package com.softicar.platform.core.module.email.mailbox.imap;
 
 import com.softicar.platform.common.core.exception.ExceptionsCollector;
-import java.util.Arrays;
+import com.softicar.platform.core.module.email.mailbox.IMailboxConnection;
+import com.softicar.platform.core.module.email.mailbox.IMailboxMessage;
 import java.util.Collection;
 import java.util.Map;
-import java.util.Properties;
 import java.util.TreeMap;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import javax.mail.Flags;
 import javax.mail.Folder;
 import javax.mail.Message;
 import javax.mail.MessagingException;
-import javax.mail.Session;
 import javax.mail.Store;
 
-public class MailboxManager implements AutoCloseable {
+class ImapConnection implements IMailboxConnection {
```

Comment:
We should have automated test coverage for this. During test execution, we could spawn a mail server, mock some things, and assert that this code does what we want it to.
Should not be all that hard, tbh - even it we need a docker container (which I doubt), we do similar things to test our SMB client (which has a _way_ larger API).
For sure, that's additional effort - but otherwise I don't see how we're protected from regressions.

Note that there also is a Java library to create an OAuth2 server:
- https://github.com/navikt/mock-oauth2-server
- https://mvnrepository.com/artifact/no.nav.security/mock-oauth2-server
We could even leverage that for the purpose of testing.

Reply:
- In general, I would agree, but I don't think we currently have the capacity to spend 1 or 2 more days to implement such tests. Also, this code was not tested before, I only adapted some superficial things that were already tested manually. Furthermore, testing oauth with office365 is hardly possible in a sandbox, you need the actual live-server.

Reply:
please confirm